posthog/models/session_replay_event/sql.py

```diff
@@ -0,0 +1,118 @@
+from django.conf import settings
+
+from posthog.clickhouse.kafka_engine import kafka_engine
+from posthog.clickhouse.table_engines import Distributed, ReplicationScheme, AggregatingMergeTree
+from posthog.kafka_client.topics import KAFKA_CLICKHOUSE_SESSION_REPLAY_EVENTS
+
+SESSION_REPLAY_EVENTS_DATA_TABLE = lambda: "sharded_session_replay_events"
+
+"""Kafka needs slightly different column setup. It receives individual events, not aggregates."""
+KAFKA_SESSION_REPLAY_EVENTS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    session_id VARCHAR,
+    team_id Int64,
+    distinct_id VARCHAR,
+    first_timestamp DateTime64(6, 'UTC'),
+    last_timestamp DateTime64(6, 'UTC'),
+    first_url Nullable(VARCHAR),
+    click_count Int64,
+    keypress_count Int64,
+    mouse_activity_count Int64
+) ENGINE = {engine}
+"""
+
+SESSION_REPLAY_EVENTS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    session_id VARCHAR,
+    team_id Int64,
+    distinct_id VARCHAR,
+    first_timestamp AggregateFunction(min, DateTime64(6, 'UTC')),
+    last_timestamp AggregateFunction(max, DateTime64(6, 'UTC')),
+    first_url Nullable(VARCHAR),
+    click_count SimpleAggregateFunction(sum, Int64),
+    keypress_count SimpleAggregateFunction(sum, Int64),
+    mouse_activity_count SimpleAggregateFunction(sum, Int64)
+) ENGINE = {engine}
+"""
+
+
+SESSION_REPLAY_EVENTS_DATA_TABLE_ENGINE = lambda: AggregatingMergeTree(
+    "session_replay_events", replication_scheme=ReplicationScheme.SHARDED
+)
+
+SESSION_REPLAY_EVENTS_TABLE_SQL = lambda: (
+    SESSION_REPLAY_EVENTS_TABLE_BASE_SQL
+    + """ORDER BY (team_id, session_id)
+SETTINGS index_granularity=512
+"""
+).format(
+    table_name=SESSION_REPLAY_EVENTS_DATA_TABLE(),
+    cluster=settings.CLICKHOUSE_CLUSTER,
+    engine=SESSION_REPLAY_EVENTS_DATA_TABLE_ENGINE(),
+)
+
+KAFKA_SESSION_REPLAY_EVENTS_TABLE_SQL = lambda: KAFKA_SESSION_REPLAY_EVENTS_TABLE_BASE_SQL.format(
+    table_name="kafka_session_replay_events",
```

Comment:
This is not for now, but we're going to be removing these Kafka Table engines to help with incident handling, maintenance tasks among other things. We'd use something like [vector](https://vector.dev/docs/reference/configuration/sinks/clickhouse/) instead, @frankh is already playing with this for the new ClickHouse cluster.

I'm just thinking about how that will work in this case 🤔  I think it's all good, we'd just be inserting into writable_session_replay_events (or maybe session_replay_events directly, handling the sharding within vector 🤔 🤷 ) ✅